guides/api-guideline.md

```diff
@@ -0,0 +1,131 @@
+# API Guideline
+
+This document outlines the best practices and conventions for designing and implementing APIs for the SDKs.
+Following these guidelines will help ensure that APIs are consistent, easy to use, and maintainable.
+Next to that the guidelines defines a syntax for documenting the APIs in a language-agnostic way for proposals.
+
+## Language Agnostic API Documentation Syntax
+
+The [proposals](../proposals) folder contains design documents of new features or changes for our SDKs.
+Each proposal should include a section that documents the API in a language-agnostic way.
+
+### Basic data types
+
+The following basic data types should be used in the API documentation.
+
+| Data Type       | Description                                                   |
+|-----------------|---------------------------------------------------------------|
+| `string`        | A sequence of characters                                      |
+| `intX`          | A signed integer of X bits                                    |
+| `uintX`         | An unsigned integer of X bits                                 |
+| `int`           | A signed integer of 32 bits                                   |
+| `long`          | A signed integer of 64 bits                                   |
```

Comment:
It seems a bit unnecessary to me to have `int` and `long` when we have `intX` and `uintX`, which are just `int32` and `int64`

Reply:
I'm still not fully convinced on not having int and long but happy to remove it for now to get the PR merged